.github/workflows/ci.yml

```diff
@@ -28,18 +28,6 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
-      - name: Gems cache
-        uses: actions/cache@v2
-        with:
-          path: ~/gems
-          key: gems-${{ matrix.ruby }}-${{ hashFiles('*.gemspec', 'Gemfile') }}-${{ github.sha }}
-          restore-keys: |
-            gems-${{ matrix.ruby }}-${{ hashFiles('*.gemspec', 'Gemfile') }}-
-            gems-${{ matrix.ruby }}-
-      - name: Install dependencies
-        run: |
-          gem install bundler
-          bundle config path ~/gems
-          bundle install --jobs 4 --retry 3
+          bundler-cache: true
       - name: Run tests
         run: bundle exec rake
```

Comment:
Maybe outside the scope of this PR, but it might be worth only running `rake spec` and not `rake rubocop` as well for every Ruby version. I don't think the results should differ between Ruby versions (@eregon might know more) and it would save time in the CI runs as well as making clear why a CI run failed.
For example, the most recent CI run failed with all Ruby versions because of RuboCop. https://github.com/prawnpdf/prawn/actions/runs/474569589

Maybe, since no `Gemfile.lock` is used to lock to a specific rubocop version, `prawn-dev` could lock it to a minor version in the `.gemspec` file instead to prevent this from happening in the first place? https://github.com/prawnpdf/prawn-dev/blob/4b370450fb10a7b7333bea35f00bc57b757987b5/prawn-dev.gemspec#L31

PS: Looking at the matrix, maybe it's also time to reconsider testing every `MAJOR.MINOR.0` version in addition to the latest patch version? I remember the [reason](https://github.com/prawnpdf/prawn/pull/1101#issuecomment-461361765) for doing it, but I wonder if the few historic events in which patch versions _introduced_ instead of _fixed_ bugs are worth expanding the matrix like that.
Of course that's up to you, @pointlessone.

Reply:
I will pin Rubocop to minor releases to make master build fail a bit less frequently. This is in the works.

I thought about making rubocop a separate job. I haven't came to a conclusion yet. I'll give it another thought.

Regarding 0 patch versions I'm not convinced. I don't see major changes in development practices. 2.7 had quite a few big changes in non-0-patch releases. We didn't test against all the patch releases and we happen to not be affected by them but it doesn't convince me that this won't happen in the future.

Right now 0-patch failures do not block merges (they're not marked "Required") and are of "informational" nature. We're nowhere near our CI limits so I'd like to keep them for now.

Reply:
> I thought about making rubocop a separate job. I haven't came to a conclusion yet. I'll give it another thought.

Is there any downside? As pros I see: 1) Don't run it 9 times with the same result. 2) Make it more obvious what failed (see `master` branch currently).

Maybe 1.1) Hypothetically, if you wanted to use rubocop's [GitHub Actions Formatter](https://docs.rubocop.org/rubocop/formatters.html#github-actions-formatter) this should probably only be run once.

> We didn't test against all the patch releases and we happen to not be affected by them but it doesn't convince me that this won't happen in the future.

I don't have a strong opinion on this and of course it's up to you, but this argument feels a little contrived to me.
What it says is, we don't test against all patch releases and nothing bad has happened with patch releases in 2.7, but because nothing happened, we should do it (but again, not every patch release). What if 0-patch versions have a bug?

I'm only wondering what we gain by testing 0-patch versions specifically vs. the latest patch version. If indeed, so many things change between patch versions, why not (exaggerated!) 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5 and 2.6.6?

But admittedly, Ruby 2.6.4 _did_ have a bug that a few people faced. It was [this string corruption bug](https://bugs.ruby-lang.org/issues/16136) I believe and we skipped it at work.

> Right now 0-patch failures do not block merges (they're not marked "Required") and are of "informational" nature. We're nowhere near our CI limits so I'd like to keep them for now.

That being said, I agree with you here and hope my words didn't sound too harsh. I'm always interested in how people approach testing! :)